library/Rules/NotEmpty.php

```diff
@@ -3,12 +3,31 @@
 
 class NotEmpty extends AbstractRule
 {
+    /** Wheter to juggle PHP types **/
+    public $jugglesTypes = true;
+
+    public function __construct($jugglesTypes = true)
+    {
+        $this->jugglesTypes = $jugglesTypes;
+    }
+
     public function validate($input)
     {
-        if (is_string($input)) {
-            $input = trim($input);
+        if (!is_string($input)) {
+            return !empty($input);
+        }
```

Comment:
Just an idea for the first major version: when the input is an array we can use `array_filter()` to remove empty values.